File: archive/d/dart/rot-13.dart
import 'dart:io';

main(List<String> args) {
  if (args.isEmpty) {
    stdout.write("Enter string to encode/decode: ");
    args = [stdin.readLineSync()];
  }

  args.forEach((arg) {
    print(rot13(arg));
  });
}

String rot13(String input) {
  int aLower = "a".codeUnitAt(0);
  int aUpper = "A".codeUnitAt(0);
  int nLower = "n".codeUnitAt(0);
  int nUpper = "N".codeUnitAt(0);
  int zLower = "z".codeUnitAt(0);
  int zUpper = "Z".codeUnitAt(0);

  List<String> coded = [];
  String codedResult;

  input.codeUnits.forEach((char) {
    if ((char >= aLower && char < nLower) ||
        (char >= aUpper && char < nUpper)) {
      coded.add(new String.fromCharCode(char + 13));
    } else if ((char >= nLower && char <= zLower) ||
        (char >= nUpper && char <= zUpper)) {
      coded.add(new String.fromCharCode(char - 13));
    } else {
      coded.add(new String.fromCharCode(char));
    }
  });

  if (coded.isNotEmpty) {
    codedResult = coded.join();
  } else {
    print("Usage: please provide a string to encrypt");
    exit(1);
  }

  return codedResult;
}

File: archive/l/lua/rot-13.lua
function mod(a, b)
    return a - (math.floor(a/b)*b)
end

if (#arg < 1)
then
    print('Usage: provide a string')
else
    str = arg[1]
    for k = 1, #str do
        local c = str:sub(k,k)
        if (c ~= " ")
        then
            io.write(string.char(mod(string.byte(c) - 97 + 13, 26) + 97))
        else
            io.write(" ")
        end
    end
end

io.write("\n")
